rllib/models/experimental/base.py

```diff
@@ -0,0 +1,142 @@
+from dataclasses import dataclass
+import abc
+
+from ray.rllib.models.specs.checker import (
+    check_input_specs,
+    check_output_specs,
+)
+from ray.rllib.models.specs.specs_dict import SpecDict
+from ray.rllib.models.temp_spec_classes import TensorDict
+from ray.rllib.utils.typing import TensorType
+from ray.rllib.utils.annotations import ExperimentalAPI
+
+ForwardOutputType = TensorDict
+STATE_IN: str = "state_in"
+STATE_OUT: str = "state_out"
+
+
+@ExperimentalAPI
+@dataclass
+class ModelConfig(abc.ABC):
+    """Configuration for a model.
+
+    Attributes:
+        output_dim: The output dimension of the network. if None, the last layer would
+            be the last hidden layer.
+    """
+
+    output_dim: int = None
+
+    @abc.abstractmethod
+    def build(self, framework: str = "torch"):
+        """Builds the model.
+
+        Args:
+            framework: The framework to use for building the model.
+        """
+        raise NotImplementedError
+
+
+class Model:
+    """Base class for RLlib models."""
```

Comment:
I feel like we need to explain here (and in the RLModule class) better the difference between `Model` (a low-level NN component that has its own input-/output specs + forward and can be added into an RLModule) and `RLModule/MARLModule` (a singular instance within an RLTrainer that contains several `Models`, depending on the algorithm and user customizations).

Something like this.

Reply:
Done!